### PR TITLE
Disallow character inputs to incidence()

### LINKED ISSUE #88: stegen not converting dates
Stegen case study: not converting dates to dates does not generate errors, but does not work either - dispatching is not working, try and see why, make sure we issue a meaningful error. 

This comes from:
```
incidence("2001-01-01")
```
Not working.

### PR DESCRIPTION
**Pull request by jrcpulliam**
Closes #88. Disallows character inputs to incidence() for the following arguments:
- dates
- first_date
- last_date

Fixes a bug where incidence() was returning an object of class Date when dates argument was character. Strict treatment of character values is intended to help prevent silent errors.

### CODE REVIEW

**Comment by zkamvar** [comment]
Hi,

I’m currently on leave for a family emergency until Mid-January. I’ll have a look at this then.

**Comment by thibautjombart** [comment]
Looking good. Will check into more details today. Many thanks!
@zkamvar this is because `as.Date("01-01-01")` does not do the job, but does not issue an error either.

File: R/incidence.R [hunk 1]
```diff
@@ -155,6 +155,9 @@ incidence <- function(dates, interval = 1L, ...) {
 #' @export
 #' @rdname incidence
 incidence.default <- function(dates, interval = 1L, ...) {
+  if (is.character(dates)) {
+    stop('Input is a character. Did you forget to convert to Date?')
+  }
   check_dates(dates)
```
Comment (thibautjombart) [inline R/incidence.R]: I would remove the `check_dates(dates)` and replace by a generic error:
```
msg <- 'Unknown date input; accepted formats are...'
stop(msg)
```
kinda stuff

File: tests/testthat/test-incidence.R [hunk 2]
```diff
@@ -225,23 +225,23 @@ test_that("corner cases", {
...
+               "last_date is a character. Did you forget to convert to Date?")
 
   expect_error(incidence(1, "week"),
                "The interval 'week' can only be used for Dates")
 
   expect_error(incidence(as.Date(Sys.Date()), standard = "TRUE"),
                "The argument `standard` must be either `TRUE` or `FALSE`")
-  
+
   expect_error(incidence(sample(10), intrval = 2),
```
Comment (thibautjombart) [inline tests/testthat/test-incidence.R]: `intrval` should be `interval`; it will be silently ignored here due to `...`

Comment (jrcpulliam) [reply tests/testthat/test-incidence.R]: 'intrval' was actually not a typo; it was a test to check the behavior for mis-spelled argument names; have left as-is

**Comment by thibautjombart** [comment]
Awesome, thanks! And welcome to the team ;)

**Comment by jrcpulliam** [comment]
My pleasure, and thanks!
